### PR TITLE
Button for mode switching

### PR DESCRIPTION
**Pull request by Tumpbe**

### CODE REVIEW

File: src/components/ProcessGraph.tsx [hunk 1]
```diff
@@ -81,18 +90,26 @@ const ProcessGraph: React.FC<ProcessGraphProps> = ({ nodes, edges, hideZoomButto
...
       }
     },
-    [selectedNode, popupTarget]
+    [selectedNode, popupTarget, selectionMode]
   );
 
+  const modeSwitch = () => {
+    setSelectionMode(!selectionMode);
+    setSelectedNode(undefined);
+    setPopupTarget(undefined);
```
Comment (vade123) [inline src/components/ProcessGraph.tsx:110]: Pull main branch and use `closePopup()`

File: src/components/ProcessGraph.tsx [hunk 2]
```diff
@@ -81,18 +90,26 @@ const ProcessGraph: React.FC<ProcessGraphProps> = ({ nodes, edges, hideZoomButto
...
   const onNodeClick = useCallback(
     (event: React.MouseEvent<SVGGElement, MouseEvent>, node: NodeData): void => {
-      const id = parseInt(node.id, 10);
-      if (id === selectedNode?.id) {
-        setSelectedNode(undefined);
-        setPopupTarget(undefined);
-      } else {
-        setSelectedNode(nodes.find(n => n.id === id));
-        setPopupTarget(event.target as Element);
+      if (!selectionMode) {
```
Comment (vade123) [inline src/components/ProcessGraph.tsx:93]: Imo better to nest less, for example in this case:

```
if (selectionMode) return;

const id ... ; 
if  ( ... ) { 
```

Or in the future 
```
if (selectionMode) someFunction();
else someOtherFunction();
```

File: src/components/ProcessGraph.tsx [hunk 3]
```diff
@@ -124,6 +141,10 @@ const ProcessGraph: React.FC<ProcessGraphProps> = ({ nodes, edges, hideZoomButto
                 visible={selectedNode !== undefined}
               />
             </TransformComponent>
+            <SwitchButton>
+              Selection mode
+              <ToggleButton value={selectionMode} onToggle={modeSwitch} />
+            </SwitchButton>
```
Comment (vade123) [inline src/components/ProcessGraph.tsx:147]: Visuality needs some work but not the most important thing at the moment :D

**Review by vade123 — COMMENTED** [review]
`.eslintrc.json` might need some additional settings:

```
"import/resolver": {
            "node": {
                  "extensions": [".js", ".jsx", ".ts", ".tsx"],
                  "moduleDirectory": ["node_modules", "src/"]
             }
}
```

**Review by vade123 — APPROVED** [review]

**Review by joonaruutiainen — APPROVED** [review]

**Pull request merged**